server/src/main/java/com/objectcomputing/checkins/services/action_item/ActionItemRepository.java

```diff
@@ -17,6 +20,12 @@
 
     List<ActionItem> findByCreatedbyid(UUID uuid);
 
+    @Query("SELECT * " +
+            "FROM action_items item " +
+            "WHERE (cast(:checkinId as uuid) IS NULL OR item.checkinid = :checkinId) " +
+            "AND (cast(:createdById as uuid) IS NULL OR item.createdbyid = :createdById)")
```

Comment:
Are you casting the right thing here? This looks backward to me. Don't you need to coerce the parameter to text (or the column to uuid) on the comparison? This looks like you are casting the already a UUID thing to a UUID on the null check.

Reply:
We have the implicit cast of uuid to varchar that is added in the flyway script, but the SQL doesn't know what to do with `NULL` when we have that. The explicit cast is needed for the null case. If you take those away then the query will work as expected when you supply a UUID (because we have an implicit cast) but blow up if you leave them null.

Reply:
actually I don't think so that we need this cast if we are using explicit cast in the migration.

Reply:
> We have the implicit cast of uuid to varchar that is added in the flyway script, but the SQL doesn't know what to do with `NULL` when we have that. The explicit cast is needed for the null case

Ahh then that make sense and that was my issue. I was not able to run a test for NULL case.